### DISCUSSION TITLE
【Notion配置不生效】SiteMap 和 RSS 不能正常读取 Notion 页面中配置的值

### DISCUSSION THREAD

**Discussion by Femoon** [tangly1024/NotionNext#2122] [Ideas]
**描述bug**
SiteMap 和 RSS 不能正确读取 Notion 页面中配置的值

**复现步骤**
https://tanghh.notion.site/8f4fe6b17a9e43e0bcfb6edb50f10a62 复制该模版
<img width="994" alt="image" src="https://github.com/tangly1024/NotionNext/assets/23479787/26f87fef-190e-4d10-8bab-3c0fe0bf3b0a">

在Notion中配置 LINK 的值，部署后 [ 域名/sitemap.xml ] 指向的是默认的 LINK 生成的 SiteMap 页面，LINK 配置不生效
<img width="522" alt="image" src="https://github.com/tangly1024/NotionNext/assets/23479787/7a37ac21-2ee3-4e3e-a60b-8b3d0066e5a0">
SiteMap 地址：https://notion-next-iqxdccnmz-femoon.vercel.app/sitemap.xml

RSS 中不能正确抓取配置中地值
<img width="722" alt="image" src="https://github.com/tangly1024/NotionNext/assets/23479787/ee514a1d-1d2a-42d7-8231-977ef986ec58">
RSS 地址：https://notion-next-iqxdccnmz-femoon.vercel.app/rss/feed.xml

**期望的正常结果**
SiteMap 和 RSS 能正确读取 Notion 页面中配置的值

**环境**

-   NotionNext版本 [4.1.1]
-   主题 [medium]
-   操作系统: [macOS]
-   浏览器 [edge]

**Comment by tangly1024** [comment]
目前暂时做不到 SiteMap 和 RSS 都读取NotionConfig，因为这两项的配置是在编译打包时生成

**Comment by dxer** [comment]
> 目前暂时做不到 SiteMap 和 RSS 都读取NotionConfig，因为这两项的配置是在编译打包时生成

大佬，我用的4.1.1版本（main分支，部署在vercel），修改 notion_config 参数都不生效，不知道是哪个步骤出了问题

**Comment by Femoon** [comment]
> 目前暂时做不到 SiteMap 和 RSS 都读取NotionConfig，因为这两项的配置是在编译打包时生成

建议在 Notion 模板中说明下，模版给了配置项但实际是不生效的

**Comment by neverrop** [comment]
> > 目前暂时做不到 SiteMap 和 RSS 都读取NotionConfig，因为这两项的配置是在编译打包时生成
> 
> 大佬，我用的4.1.1版本（main分支，部署在vercel），修改 notion_config 参数都不生效，不知道是哪个步骤出了问题

@dxer @tangly1024 我也一样，在notion config里改theme啥的都没用，用的最新的4.1.3，你后来有知道原因吗。
<img width="663" alt="Screenshot 2024-01-09 at 07 11 29" src="https://github.com/tangly1024/NotionNext/assets/12857073/94e1d439-9856-4377-8c3d-c07c78ca4e57">

**Comment by niustark** [comment]
思维

> > > 目前暂时做不到 SiteMap 和 RSS 都读取NotionConfig，因为这两项的配置是在编译打包时生成
> > 
> > 
> > 大佬，我用的4.1.1版本（main分支，部署在vercel），修改 notion_config 参数都不生效，不知道是哪个步骤出了问题
> 
> @dxer @tangly1024 我也一样，在notion config里改theme啥的都没用，用的最新的4.1.3，你后来有知道原因吗。 <img alt="Screenshot 2024-01-09 at 07 11 29" width="663" src="https://private-user-images.githubusercontent.com/12857073/295059157-94e1d439-9856-4377-8c3d-c07c78ca4e57.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxMDg0MzAsIm5iZiI6MTcwNjEwODEzMCwicGF0aCI6Ii8xMjg1NzA3My8yOTUwNTkxNTctOTRlMWQ0MzktOTg1Ni00Mzc3LThjM2QtYzA3Yzc4Y2E0ZTU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI0VDE0NTUzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0OTY3YzhiYzdiMjdmYjk5ZWNlYmY3YjZiNzU5ZmFlOTA4MjZjZDQ0NWQyZmE2YjQ0OWExNTEyMWJhYWQ3YzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.L2yaZXyFrWUwzJFeoB-wuWg9pyTSXYWsSXpCq8WinQU">
我也发现个问题 
notion里的config配置了不起作用,还是得在代码里配置NotionNext /themes/heo/config.js
heo主题配置文件里的 链接内容不能为空,
HEO_HERO_TITLE_LINK:
HEO_INFO_CARD_URL:

**Comment by xiaoyuuuuuupeng** [comment]
> > > 目前暂时做不到 SiteMap 和 RSS 都读取NotionConfig，因为这两项的配置是在编译打包时生成
> > 
> > 
> > 大佬，我用的4.1.1版本（main分支，部署在vercel），修改 notion_config 参数都不生效，不知道是哪个步骤出了问题
> 
> @dxer @tangly1024 我也一样，在notion config里改theme啥的都没用，用的最新的4.1.3，你后来有知道原因吗。 <img alt="Screenshot 2024-01-09 at 07 11 29" width="663" src="https://private-user-images.githubusercontent.com/12857073/295059157-94e1d439-9856-4377-8c3d-c07c78ca4e57.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2Njk3MDIsIm5iZiI6MTcwNjY2OTQwMiwicGF0aCI6Ii8xMjg1NzA3My8yOTUwNTkxNTctOTRlMWQ0MzktOTg1Ni00Mzc3LThjM2QtYzA3Yzc4Y2E0ZTU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDAyNTAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxNzFhNGMwODA2MWRiOTdiZWQ5ODNiNjA2Nzg1ZGIyNzNhMGQyZjFkY2U3MjYyNjJiYzFjODEyMTg5Mjk2YzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TOlaYRUt1aDrtEfwEhSxV_GXvuWMTFxKL7mG48VBUuA">

我也是这样的 主题配置没有用 大佬解决了吗
